lib/util/Components.js

```diff
@@ -787,6 +789,93 @@ function componentRule(rule, context) {
         && !!(node.params || []).length
       );
     },
+
+    /**
+     * Identify whether a node (CallExpression) is a call to a React hook
+     *
+     * @param {ASTNode} node The AST node being searched. (expects CallExpression)
+     * @param {('useCallback'|'useContext'|'useDebugValue'|'useEffect'|'useImperativeHandle'|'useLayoutEffect'|'useMemo'|'useReducer'|'useRef'|'useState')[]} [expectedHookNames] React hook names to which search is limited.
+     * @returns {Boolean} True if the node is a call to a React hook
+     */
+    isReactHookCall(node, expectedHookNames) {
+      if (node.type !== 'CallExpression') {
+        return false;
+      }
+
+      const defaultReactImports = components.getDefaultReactImports();
+      const namedReactImports = components.getNamedReactImports();
+
+      const defaultReactImportSpecifier = defaultReactImports
+        ? defaultReactImports[0]
+        : undefined;
+
+      const defaultReactImportName = defaultReactImportSpecifier
+        ? defaultReactImportSpecifier.local.name
+        : undefined;
+
+      const reactHookImportSpecifiers = namedReactImports
+        ? namedReactImports.filter((specifier) => specifier.imported.name.match(USE_HOOK_PREFIX_REGEX))
+        : undefined;
+      const reactHookImportNames = reactHookImportSpecifiers
+        ? reactHookImportSpecifiers.reduce(
+          (acc, specifier) => {
+            acc[specifier.local.name] = specifier.imported.name;
+            return acc;
+          },
+          {}
+        )
+        : undefined;
```

Comment:
```suggestion
      const reactHookImportNames = reactHookImportSpecifiers
        && fromEntries(reactHookImportSpecifiers.map(({ local, imported }) => [local.name, imported.name]);
```
and pull in `object.fromentries` for fromEntries

Reply:
whoops, oh well, we can just name the argument then